### PR TITLE
Project Enumerable code review request!!

### PR DESCRIPTION
**Pull request by Cyrus-Kiprop**
This project involves the recreation of the Ruby's major Enumerable methods found in its Enumerable module. The project includes a detailed  readme and a main.rb file that houses all the logic .  Please do review it and feel free to point any areas that might require improvement. 
Thank you.

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+      end
+    else
+      to_a.my_each do |item|
+        return true if item
+      end
+    end
+    false
+  end
+
+  def my_none?(*args)
```
Comment (Forison) [inline main.rb:87]: - ```my_none?``` and ```none?``` must return same results in the examples below
```%w{ant bear cat}.my_none?{ |word| word.length >=4}  != %w{ant bear cat}.none?{ |word| word.length >= 4 }```
``` [1, 3.14, 42].my_none?(Float) != [1, 3.14, 42].none?(Float)```
``` [ ].my_none? `!=` [ ].none? ```
``` [nil, false, true].my_none? `!=`  [nil, false, true].none? ```

File: main.rb [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+      end
+    else
+      to_a.my_each do |item|
+        return true unless item
+      end
+    end
+    false
+  end
+
+  def my_count(*args)
```
Comment (Forison) [inline main.rb:114]: - compare results in both cases

```ary = [1, 2, 4, 2]
   ary.my_count(2, &:even?) != ary.count(2, &:even?)
```
I think the right behavior here ought to be tested by ```ary.count(&:even?)```
 in our case ```ary.my_count(&:even?)``` I am raising this issue for you to go back to the functionalities of the method so you can be clear on it.

**Review by Forison — CHANGES_REQUESTED** [review]
### Status: Require Changes🛑
# 🙌Kiprop,
most of your custom methods are passing the test cases🙂. Your code is clean but there are few methods that need to be corrected🙂.
* [ ] should have working my_none? identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F) ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
* [ ] should have working `my_count` identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
## Check my comment below 👇 for clarifications
### That is it kindly make these changes and submit the project for review🙂
### Happy coding💻
#### Addo Forison

File: main.rb (hunk 1 above)
Comment (Cyrus-Kiprop) [reply main.rb:87]: Thanks for your review and time @Forison. Hope you are having an awesome time. I took my time and indeed found some unattended edge cases on the methods you mentioned above. I went further and fixed the bugs and tested it. I can firmly say that the methods are working as expected.
 Thank you.

**Review by jofer86 — CHANGES_REQUESTED** [review]
## :wave: :wave: Hello!, I'm Jorge, and  I'm here to review this Pull Request, let's take a look shall we :eye: :eye: :eye: 
- Changes marked as **[OPTIONAL]** are not hard requirements but are nice to have.
- There are still issues on your code, mainly on your `my_none?` method and your `my_inject` method, here I'll leave a screenshot of the issue for further understanding.
My_none method, there's the ruby method and then your method with the corresponding output
image
- And for my inject: 
image

- Take some time and re think what could've gone wrong there, and do ask for another code review please.

--------------------------------------------------------------------
Cheers, 
Jorge Fernando Rincón Figueroa.
Technical Support Engineer.
**Slack: @jofer86**

**Comment by Cyrus-Kiprop** [comment]
Thanks for your time @Jorge. I have investigated the matter and I am working on it. I did copy paste the test cases from rubys official documentation to taste my code. It turns out that I didn't rename the none? into my_none? method. I am totally suprised how my algorithm is failing at all cases. This is hillarious. Thank Jorge. Working on the issue asap.

**Comment by Cyrus-Kiprop** [comment]
Oh and by the way your review was lovely. I totally appreciate it. 
Made me realize some stupid mistakes I did while coding.

**Review by stratospherique — APPROVED** [review]
### Great Job applying the suggested changes :clap: :clap: Congrats, the project is approved :heavy_check_mark: You may merge :+1: :+1: 

Keep up the good work :muscle: 
**_Ahmed Mahfoudh_**

**Pull request merged**